### PR TITLE
Kernel: T5626: only select required Kernel CGROUP controllers (backport)

### PR DESCRIPTION
**Pull request by c-po**
## Change Summary

TL;DR: systemd does not require the performance-sensitive bits of Linux control
groups enabled in the kernel. However, it does require some
non-performance-sensitive bits of the control group logic.

http://0pointer.de/blog/projects/cgroups-vs-cgroups.html

The only controllers required for VyOS to function are the memory and PID
controller required by the container feature. All other controlles can be
disabled.

(cherry picked from commit 86f80f99c57240bda43b99d110de736aef4dea76)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [x] Other (please describe): cleanup and performance

## Related Task(s)

* https://vyos.dev/T5626

## Component(s) name

Kernel

## How to test

Smoketests

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [x] I have linked this PR to one or more Phabricator Task(s)
- [x] My commit headlines contain a valid Task id
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

**Review by sever-sever — APPROVED** [review]

**Comment by Apachez-** [comment]
My €0.05:

It seems like the updated kernel config includes changes not described by this PR.

For example these lines who have changed from "m" to "y"?

```
CONFIG_ZSTD_COMMON=y
CONFIG_ZSTD_DECOMPRESS=y
```

**Comment by c-po** [comment]
> My €0.05:
> 
> It seems like the updated kernel config includes changes not described by this PR.
> 
> For example these lines who have changed from "m" to "y"?
> 
> ```
> CONFIG_ZSTD_COMMON=y
> CONFIG_ZSTD_DECOMPRESS=y
> ```

See commit 478d7e7df846b457115f7d429792470675dde5d8

So this only versions control the Kernel config with options set implicitly during the build process.

**Review by jestabro — APPROVED** [review]
